frame/pablo/src/lib.rs

```diff
@@ -467,6 +546,104 @@ pub mod pallet {
+                            let (
+                                (elapsed, base_price_cumulative),
+                                (_, quote_price_cumulative),
+                            ) = (
+                            compute_next_price_cumulative::<T::Convert, _, _>(
+                                previous_cumulative_price.timestamp,
+                                previous_cumulative_price.base_price_cumulative,
+                                current_timestamp,
+                                rate_base
+                            )?,
+                            compute_next_price_cumulative::<T::Convert, _, _>(
+                                previous_cumulative_price.timestamp,
+                                previous_cumulative_price.quote_price_cumulative,
+                                current_timestamp,
+                                rate_quote
+                            )?,
+                            );
+                            let twap_interval = T::TWAPInterval::get();
+                            if elapsed >= twap_interval {
+                                // PriceCumulativesState::<T>::insert(
+                                //     pool_id,
+                                //     PriceCumulatives {
+                                //         timestamp: current_timestamp,
+                                //         base_price_cumulative,
+                                //         quote_price_cumulative,
+                                //     }
+                                // );
+                                let avg = |current: T::Balance,
+                                previous: T::Balance|
+                                    -> Result<Rate, DispatchError> {
+                                        compute_price_average::<T::Convert, _, _>(
+                                            current, previous, elapsed,
+                                        )
+                                    };
+                                let average_price_base = avg(
+                                    base_price_cumulative,
+                                    previous_cumulative_price.base_price_cumulative,
+                                )?;
+                                let average_price_quote = avg(
+                                    quote_price_cumulative,
+                                    previous_cumulative_price.quote_price_cumulative,
+                                )?;
+                                TWAPState::<T>::insert(
+                                    pool_id,
+                                    TimeWeightedAveragePrice {
+                                        timestamp: current_timestamp,
+                                        base_price_cumulative,
+                                        quote_price_cumulative,
+                                        base_twap: average_price_base,
```

Comment:
average_price_base can be found from base_price_cumulative. why we store it?

Reply:
IMHO it should not matter much, with storage we get some saving on execution and if we want  to save execution time we use small bit of storage.

Reply:
But that price is not average. It is last price value measured. What us use what last price? It may be used one in the start of block, but than it changes and need to be calculated.

Reply:
isn't it should be calculated after every TWAPInternal time?